source/patterns/@aws-solutions-constructs/aws-iot-s3/README.md

```diff
@@ -64,6 +64,7 @@ _Parameters_
 |loggingBucketProps?|[`s3.BucketProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.BucketProps.html)|Optional user provided props to override the default props for the S3 Logging Bucket.|
 |iotTopicRuleProps?|[`iot.CfnTopicRuleProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iot.CfnTopicRuleProps.html)|User provided CfnTopicRuleProps to override the defaults.|
 |s3Key|`string`|User provided s3Key to override the default (`${topic()}/${timestamp()}`) object key. Used to store messages matched by the IoT Rule.|
+|logS3AccessLogs?|`boolean`|Whether to turn on Access Logging for the S3 bucket. Creates an S3 bucket with associated storage costs for the logs. Enabling Access Logging is a best practice. default - true|
 
 ## Pattern Properties
 
```

Comment:
Properties needs to expose s3BucketInterface: s3:IBucket as well. You accept IBucket as an input - if the client sends an IBucket, how do you populate this non-optional property with an s3.Bucket? The old constructs aren't consistent on this, but where we want to be is a construct accepts an IBucket or Bucket as a prop (but not both).

If a construct accepts IBucket, then it exposes Bucket and IBucket at properties. If it receives an IBucket then it populates just s3BucketInterface. If it creates a new bucket, then it populates BOTH s3BucketInterface and s3Bucket.

If a construct accepts Bucket, then we expose just s3Bucket and always populate it.